client/my-sites/current-site/index.jsx

```diff
@@ -141,7 +142,7 @@ export default connect(
 	state => ( {
 		selectedSite: getSelectedSite( state ),
 		anySiteSelected: getSelectedOrAllSites( state ),
-		siteCount: getVisibleSites( state ).length,
+		visibleSitesCount: getCurrentUserVisibleSiteCount( state ),
```

Comment:
What if this change how the site count is retrieved was an independent PR? Making this whole refactoring work will need some effort, and spinning off small standalone PRs will be helpful.

Reply:
Yes, I plan to split this PR into multiple smaller ones. Then try to get them in one by one, so we'll have most of the functionality ready. Finally, we just switch the toggle. :)

However, I like prototyping in one PR, it's much easier to work that way, especially if you don't know what all you need to touch.